### ISSUE TITLE
Character configuration values need to be validated before being stored in configuration file

### ISSUE THREAD

**Issue by derekpierre** [nucypher/nucypher#2661]
**Describe the Bug**
I am able to successfully write non-sensical values to an ursula configuration file because the values aren't validated:

```bash
nucypher ursula config --provider https://rinkeby.infura.io/v3/fake --rest-host 22.22.23.123 --network mainnet_fake --signer keystore:///Users/derek/Library/Ethereum/rinkeby/fake --config-file "/Users/derek/Library/Application Support/nucypher/ursula.json"

```
This successfully works for updating the values in the `ursula.json` configuration file.

However, if I try to update the values to be correct, then validation of the values occur and I can't update the file via the CLI, needing to resort to modifying the file directly using an editor (current workaround). Examples of tracebacks:
```
ValueError: GithubRegistrySource not available for network 'mainnet_fake'. Valid options are: ['mainnet', 'ibex', 'lynx']
```
After manually updating "domain" value in config file:
```
nucypher.blockchain.eth.signers.base.InvalidSignerURI: No such keystore file or directory "/Users/derek/Library/Ethereum/rinkeby/keystore/fake"
```
After manually updating keystore signer path the config file could be updated

Basically we need to perform validation on values used to write to the configuration file eg. ensure network value is valid, or keystore signer file exists, or...
